aviatrix/resource_aviatrix_transit_gateway_test.go

```diff
@@ -362,39 +445,81 @@ resource "aviatrix_transit_gateway" "test_transit_gateway_aep" {
 	account_name = aviatrix_account.test_acc_edge_aep.account_name
 	gw_name      = "tfg-edge-aep-%[1]s"
 	vpc_id       = "%[2]s"
-	site_id 	= "%[2]s"
 	device_id = "%[3]s"
 	gw_size      = "SMALL"
 	interfaces {
-        gateway_ip = "192.168.24.1"
-        ifname     = "eth0"
-        ipaddr    = "192.168.24.13/24"
-        type       = "WAN"
+        gateway_ip    = "192.168.20.1"
+        ip_address    = "192.168.20.11/24"
+        type          = "WAN"
+        index         = 0
+        secondary_private_cidr_list = ["192.168.19.16/29"]
     }
+
     interfaces {
-        gateway_ip = "192.168.13.1"
-        ifname     = "eth1"
-        ipaddr    = "192.168.13.33/24"
-        type       = "WAN"
+        gateway_ip    = "192.168.21.1"
+        ip_address    = "192.168.21.11/24"
+        type          = "WAN"
+        index         = 1
+        secondary_private_cidr_list = ["192.168.21.16/29"]
     }
+
     interfaces {
         dhcp   = true
-        ifname = "eth2"
         type   = "MANAGEMENT"
+        index  = 0
     }
+
     interfaces {
-        gateway_ip                  = "192.168.19.1"
-        ifname                      = "eth3"
-        ipaddr                     = "192.168.19.13/24"
-        type                        = "WAN"
-        secondary_private_cidr_list = ["192.168.19.112/29"]
+        gateway_ip  = "192.168.22.1"
+        ip_address  = "192.168.22.11/24"
+        type        = "WAN"
+        index       = 2
     }
+
     interfaces {
-        gateway_ip                  = "192.168.18.1"
-        ifname                      = "eth4"
-        ipaddr                     = "192.168.18.13/24"
-        type                        = "WAN"
-        secondary_private_cidr_list = ["192.168.18.112/29"]
+        gateway_ip = "192.168.23.1"
+        ip_address = "192.168.23.11/24"
+        type       = "WAN"
+        index      = 3
+    }
+
+	ha_device_id = "a20c75c0-06c2-4102-9df1-b00b85e89eac"
+    peer_backup_port_type = "WAN"
```

Comment:
I think this is perfectly fine.
But based on some of the discussions today, people were suggesting to use an attibute combining both type and index like:
LAN0, WAN1 etc.
I understand that is like this for interface config, specified separately.
But for other pieces of config, just get Shiva's opinion as well or if that can be done later, then having this currently is fine as well.

Reply:
I think that will require some significant code change in the getter and setter function. Can include that change based on what Shiva has to say.